Build option PICO_TOOLCHAIN_PATH has no effect

Was trying to use `PICO_TOOLCHAIN_PATH` to point to the location where I have it installed on my setup.
(I do _NOT_ have gcc it on my path)

Turns out that this option has no effect because it's always cleared [here](https://github.com/raspberrypi/pico-sdk/blob/26653ea81e340cacee55025d110c3e014a252a87/cmake/pico_pre_load_toolchain.cmake#L3).

As soon as I comment out this line, I can set the `PICO_TOOLCHAIN_PATH` option and `pico_find_compiler` happilly finds the compiler.

I can submit a PR with this fix if you want.

Follows the CMake logs, for completeness. On both the `PICO_TOOLCHAIN_PATH` is set trough `.vscode\cmake-variants.json`. Build folder cleaned.

With the original code:

```
[main] Configuring folder: pico-sdk 
[proc] Executing command: "C:/Program Files/CMake/bin/cmake.exe" --no-warn-unused-cli -DCMAKE_MAKE_PROGRAM:STRING=E:/nftools/Ninja/ninja.exe -DPICO_TOOLCHAIN_PATH:STRING=E:/nftools/GNU_Tools_ARM_Embedded/9-2020-q2-update/ -DPICO_COMPILER:STRING=pico_arm_gcc -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -He:/Github/pico-sdk -Be:/Github/pico-sdk/build -G Ninja
[cmake] Not searching for unused variables given on the command line.
[cmake] PICO_SDK_PATH is E:/Github/pico-sdk
[cmake] Defaulting PICO_PLATFORM to rp2040 since not specified.
[cmake] PICO compiler is pico_arm_gcc
[cmake] PICO_GCC_TRIPLE defaulted to arm-none-eabi
[cmake] CMake Error at cmake/preload/toolchains/find_compiler.cmake:28 (message):
[cmake]   Compiler 'arm-none-eabi-gcc' not found, you can specify search path with
[cmake]   "PICO_TOOLCHAIN_PATH".
[cmake] Call Stack (most recent call first):
[cmake]   cmake/preload/toolchains/pico_arm_gcc.cmake:20 (pico_find_compiler)
[cmake]   C:/Program Files/CMake/share/cmake-3.15/Modules/CMakeDetermineSystem.cmake:93 (include)
[cmake]   CMakeLists.txt:6 (project)
[cmake] 
[cmake] 
[cmake] CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
[cmake] CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
[cmake] CMake Error: CMAKE_ASM_COMPILER not set, after EnableLanguage
[cmake] -- Configuring incomplete, errors occurred!
```

With line 3 @ pico_pre_load_toolchain.cmake commented:

```
[cmake] -- Build files have been written to: E:/Github/pico-sdk/build
[main] Configuring folder: pico-sdk 
[proc] Executing command: "C:/Program Files/CMake/bin/cmake.exe" --no-warn-unused-cli -DCMAKE_MAKE_PROGRAM:STRING=E:/nftools/Ninja/ninja.exe -DPICO_TOOLCHAIN_PATH:STRING=E:/nftools/GNU_Tools_ARM_Embedded/9-2020-q2-update/ -DPICO_COMPILER:STRING=pico_arm_gcc -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -He:/Github/pico-sdk -Be:/Github/pico-sdk/build -G Ninja
[cmake] Not searching for unused variables given on the command line.
[cmake] PICO_SDK_PATH is E:/Github/pico-sdk
[cmake] Defaulting PICO_PLATFORM to rp2040 since not specified.
[cmake] PICO compiler is pico_arm_gcc
[cmake] PICO_GCC_TRIPLE defaulted to arm-none-eabi
[cmake] -- The C compiler identification is GNU 9.3.1
[cmake] -- The CXX compiler identification is GNU 9.3.1
[cmake] -- The ASM compiler identification is GNU
[cmake] -- Found assembler: E:/nftools/GNU_Tools_ARM_Embedded/9-2020-q2-update/bin/arm-none-eabi-gcc.exe
[cmake] Using regular optimized debug build (set PICO_DEOPTIMIZED_DEBUG=1 to de-optimize)
[cmake] Defaulting PICO target board to pico since not specified.
[cmake] Using board configuration from E:/Github/pico-sdk/src/boards/include/boards/pico.h
[cmake] -- Found Python3: C:/Program Files/Python39/python.exe (found version "3.9.2") found components:  Interpreter 
[cmake] TinyUSB available at E:/Github/pico-sdk/lib/tinyusb/src/portable/raspberrypi/rp2040; adding USB support.
[cmake] Compiling TinyUSB with CFG_TUSB_DEBUG=1
[cmake] ELF2UF2 will need to be built
[cmake] -- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
[cmake] -- Configuring done
[cmake] -- Generating done
[cmake] -- Build files have been written to: E:/Github/pico-sdk/build
```

--- a/cmake/pico_pre_load_toolchain.cmake
+++ b/cmake/pico_pre_load_toolchain.cmake
@@ -1,6 +1,5 @@
 # PICO_CMAKE_CONFIG: PICO_TOOLCHAIN_PATH, Path to search for compiler, default=none (i.e. search system paths), group=build
-# Set your compiler path here if it's not in the PATH environment variable.
-set(PICO_TOOLCHAIN_PATH "" CACHE INTERNAL "")
+set(PICO_TOOLCHAIN_PATH "${PICO_TOOLCHAIN_PATH}" CACHE INTERNAL "")
 
 # Set a default build type if none was specified
 set(default_build_type "Release")

